### ISSUE TITLE
Default paper size is owerwritten in PDF render function

### ISSUE THREAD

**Issue by lkreher** [barryvdh/laravel-dompdf#515]
In [PDF.php](https://github.com/barryvdh/laravel-dompdf/blob/master/src/PDF.php) line 198 `$this->dompdf->setPaper($this->paper, $this->orientation);`, `$this->paper` is null by default, causing the default_paper_size configuration to be ignored. Removing the line fixes the problem.

**Comment by lkreher** [comment]
> default paper is 'portrait' dude
check thifunction
public function setPaper($paper, $orientation = 'portrait'){
  | $this->paper = $paper;
  | $this->orientation = $orientation;
  | $this->dompdf->setPaper($paper, $orientation);
  | return $this;
  | }

Default orientation is 'portrait', but there is no default paper, and the function call sets `$this->paper` to the previous value of `$this->paper` which is null.

**Comment by foufos** [comment]
The value of `$this->paper` is null, so it falls back to the default setting of the @dompdf package which is `private $defaultPaperSize = "letter";`
[dompdf default options](https://github.com/dompdf/dompdf/blob/6f9ff4b084373bb934b7cbe2cc31a83ddf1bb2a7/src/Options.php#L91)

So it's just an issue with the default docblock definition of the [setPaper](https://github.com/barryvdh/laravel-dompdf/blob/46781d0304277845a19c09c169bc595fd182cce4/src/PDF.php#L63)
It states "Set the paper size (default A4)", when it should be "Set the paper size (default letter)"

**Comment by lkreher** [comment]
> The value of `$this->paper` is null, so it falls back to the default setting of the @dompdf package which is `private $defaultPaperSize = "letter";`
> [dompdf default options](https://github.com/dompdf/dompdf/blob/6f9ff4b084373bb934b7cbe2cc31a83ddf1bb2a7/src/Options.php#L91)
> 
> So it's just an issue with the default docblock definition of the [setPaper](https://github.com/barryvdh/laravel-dompdf/blob/46781d0304277845a19c09c169bc595fd182cce4/src/PDF.php#L63)
> It states "Set the paper size (default A4)", when it should be "Set the paper size (default letter)"

No, the issue is, the `config/dompdf.php` file has a field to configure `default_paper_size`, but this config value is always overwritten by "letter" unless a paper size is explicitly specified when calling the PDF generator. 

Basically right now the order is _setPaper()_ > _fallback value used by dompdf ("letter")_ > _config value of `default_paper_size`_ when the last two should be switched around, so config value is used over fallback value.

By loading the `config/dompdf.php` the paperSize is actually correctly saved to the dompdf instance, but when performing the laravel-dompdf `render()` function there is an explicit call to `$this->dompdf->setPaper()` which resets the paper to `null`, thus causing dompdf to use the fallback again. 

[PDF.php](https://github.com/barryvdh/laravel-dompdf/blob/master/src/PDF.php) line 198 evaluates to `$this->dompdf->setPaper(null, null)` resetting the dompdf instance to its default values just before actually rendering the PDF.

**Comment by djereg** [comment]
@lkreher you're right! Today I was ran into this error.

Right now the solution to my problem is settings the paper size (instead of the default value) before calling output function.
